build-logic/src/main/kotlin/PublicationPlugin.kt

```diff
@@ -0,0 +1,126 @@
+                                    distribution.set("repo")
+                                }
+                            }
+                            developers {
+                                developer {
+                                    id.set(providers.gradleProperty("POM_DEVELOPER_ID"))
+                                    name.set(providers.gradleProperty("POM_DEVELOPER_NAME"))
+                                }
+                            }
+                            scm {
+                                url.set(providers.gradleProperty("POM_SCM_URL"))
+                                connection.set(providers.gradleProperty("POM_SCM_CONNECTION"))
+                                developerConnection.set(providers.gradleProperty("POM_SCM_DEV_CONNECTION"))
+                            }
+                        }
+                    }
+                }
+
+                repositories {
+                    maven {
+                        val releasesRepoUrl =
+                            URI.create(
+                                "https://ossrh-staging-api.central.sonatype.com/service/local/staging/deploy/maven2/",
+                            )
+                        val snapshotsRepoUrl = URI.create("https://central.sonatype.com/repository/maven-snapshots/")
+                        name = "mavenCentral"
+                        url = if (version.toString().endsWith("SNAPSHOT")) snapshotsRepoUrl else releasesRepoUrl
+
+                        logger.info("Set publication repository for version $version to $url")
+
+                        credentials {
+                            username = providers.gradleProperty("CENTRAL_PORTAL_USERNAME").orNull
+                                ?: System.getenv("CENTRAL_PORTAL_USERNAME")
+                            password = providers.gradleProperty("CENTRAL_PORTAL_TOKEN").orNull
+                                ?: System.getenv("CENTRAL_PORTAL_TOKEN")
+                        }
+                    }
+
+                    maven {
+                        name = "mavenCentralRelocation"
+                        url = URI.create("https://oss.sonatype.org/service/local/staging/deploy/maven2/")
+
+                        credentials {
+                            username = providers.gradleProperty("CENTRAL_PORTAL_USERNAME").orNull
+                                ?: System.getenv("CENTRAL_PORTAL_USERNAME")
+                            password = providers.gradleProperty("CENTRAL_PORTAL_TOKEN").orNull
+                                ?: System.getenv("CENTRAL_PORTAL_TOKEN")
+                        }
+                    }
+                }
+            }
+
+        /*
+         * Following signing parameters must be configured in `$HOME/.gradle/gradle.properties`:
+         * ```properties
+         * signing.keyId=12345678
+         * signing.password=some_password
+         * signing.secretKeyRingFile=/Users/yourusername/.gnupg/secring.gpg
+         * ```
+         */
```

Comment:
Indentation of this block is not correct. This should have been caught by ktlint format.